cpp/tests/experimental/spatial/linestring_bounding_boxes_test.cu

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cuspatial_test/vector_equality.hpp>
+#include <cuspatial_test/vector_factories.cuh>
+
+#include <cuspatial/error.hpp>
+#include <cuspatial/experimental/geometry/box.hpp>
+#include <cuspatial/experimental/linestring_bounding_boxes.cuh>
+#include <cuspatial/vec_2d.hpp>
+
+#include <gtest/gtest.h>
+
+template <typename T>
+struct LinestringBoundingBoxTest : public ::testing::Test {
+};
+
+using cuspatial::vec_2d;
+using cuspatial::test::make_device_vector;
+
+using TestTypes = ::testing::Types<float, double>;
+
+TYPED_TEST_CASE(LinestringBoundingBoxTest, TestTypes);
+
+TYPED_TEST(LinestringBoundingBoxTest, test_empty)
+{
+  using T = TypeParam;
+
+  auto offsets  = make_device_vector<int32_t>({});
+  auto vertices = make_device_vector<vec_2d<T>>({});
+
+  auto bboxes = rmm::device_vector<cuspatial::box<T>>(offsets.size());
+
+  auto bboxes_end = cuspatial::linestring_bounding_boxes(
+    offsets.begin(), offsets.end(), vertices.begin(), vertices.end(), bboxes.begin());
+
+  EXPECT_EQ(std::distance(bboxes.begin(), bboxes_end), 0);
+}
+
+TYPED_TEST(LinestringBoundingBoxTest, test_one)
+{
+  using T = TypeParam;
+
+  auto offsets  = make_device_vector<int32_t>({0});
+  auto vertices = make_device_vector<vec_2d<T>>(
+    {{2.488450, 5.856625}, {1.333584, 5.008840}, {3.460720, 4.586599}, {2.488450, 5.856625}});
+
+  auto bboxes = rmm::device_vector<cuspatial::box<T>>(offsets.size());
+
+  auto bboxes_end = cuspatial::linestring_bounding_boxes(
+    offsets.begin(), offsets.end(), vertices.begin(), vertices.end(), bboxes.begin());
+
+  EXPECT_EQ(std::distance(bboxes.begin(), bboxes_end), 1);
+
+  auto bboxes_expected =
+    make_device_vector<cuspatial::box<T>>({{{1.333584, 4.586599}, {3.460720, 5.856625}}});
+
+  CUSPATIAL_EXPECT_VECTORS_EQUIVALENT(bboxes, bboxes_expected);
+}
+
+TYPED_TEST(LinestringBoundingBoxTest, test_small)
+{
+  using T = TypeParam;
+
+  auto offsets  = make_device_vector<int32_t>({0, 4, 9, 13});
```

Comment:
Same concern here, users coming from a `GeoSeries` will need to slice off the last value with `offset[:-1]` in order to use the API.